cdk/dcm_floodlight/transform.js

```diff
@@ -0,0 +1,171 @@
+/* eslint-disable no-param-reassign */
+const get = require("get-value");
+const {
+  getValueFromMessage,
+  removeUndefinedAndNullValues,
+  isDefinedAndNotNull
+} = require("rudder-transformer-cdk/build/utils");
+const { CustomError, getIntegrationsObj, isEmpty } = require("../../v0/util");
+
+// append properties to endpoint
+// eg: ${endpoint}key1=value1;key2=value2;....
+function appendProperties(endpoint, payload) {
+  Object.keys(payload).forEach(key => {
+    endpoint += `${key}=${payload[key]};`;
+  });
+
+  return endpoint;
+}
+
+// transform webapp dynamicForm custom floodlight variable
+// into {u1: value, u2: value, ...}
+function transformCustomVariable(customFloodlightVariable) {
+  const customVariable = {};
+  customFloodlightVariable.forEach(item => {
+    if (!isEmpty(item.from) && !isEmpty(item.to)) {
+      // remove u if already there
+      customVariable[
+        `u${item.from.trim().replace(/u/g, "")}`
+      ] = `[${item.to.trim()}]`;
+    }
+  });
+
+  return customVariable;
+}
+
+// valid flag should be provided [1|true] or [0|false]
+function isValidFlag(key, value) {
```

Comment:
We should consider value as undefined case

Reply:
If you check the condition before calling this function it has been handled. We don't require to double-check it.

Reply:
Agreed i we have a check before the function call on what we are passing as value is valid its redundant code